File: Android-VS-iOS/Andriod_and_iOS_test_cases/equationSolver/Android/MainActivity.java
package com.example.david.functionsolver;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private double a_value;
    private double b_value;
    private double c_value;
    EditText editTextA;
    EditText editTextB;
    EditText editTextC;
    TextView x1TextView;
    TextView x2TextView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        x1TextView = findViewById(R.id.x1_value_textView);
        x2TextView = findViewById(R.id.x2_value_textView);
        editTextA = findViewById(R.id.a_ValueEditText);
        editTextA.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }
            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }
            @Override
            public void afterTextChanged(Editable editable) {
                if (editable.length() != 0)
                    a_value = Double.parseDouble(editable.toString());
                else {
                    a_value = 0;
                }
                setNewValues();
            }
        });
        editTextB = findViewById(R.id.b_ValueEditText);
        editTextB.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void afterTextChanged(Editable editable) {
                if (editable.length() != 0)
                    b_value = Double.parseDouble(editable.toString());
                else {
                    b_value = 0;
                }
                setNewValues();

            }
        });
        editTextC = findViewById(R.id.c_ValueEditText);
        editTextC.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void afterTextChanged(Editable editable) {
                if (editable.length() != 0)
                    c_value = Double.parseDouble(editable.toString());
                else {
                    c_value = 0;
                }
                setNewValues();

            }
        });
    }
    private void setNewValues() {
        double determinant = Math.pow(b_value,2) - 4*a_value*c_value;
        String x1 , x2;
        if(determinant > 0){
            double temp1 = -b_value/(2*a_value);
            double temp2 = Math.sqrt(determinant)/(2*a_value);
            x1 = String.valueOf((temp1+temp2));
            x2 = String.valueOf(temp1-temp2);
        }else if (determinant == 0){
            x1 = String.valueOf(-b_value/(2*a_value));
            x2 = x1;

        }
        else{
            double temp1 =  -b_value/(2*a_value);
            double temp2 =  Math.sqrt(-determinant)/(2*a_value);
            x1 =  String.valueOf(temp1) + "+i" +
                    String.valueOf(temp2);

            x2 =  String.valueOf(temp1) + "-i" +
                    String.valueOf(temp2);
        }

        x1TextView.setText(x1);
        x2TextView.setText(x2);

    }




}

File: Android-VS-iOS/Andriod_and_iOS_test_cases/equationSolver/iOS/MainActivity.swift
import UIKit


 class MainActivity: UIViewController{
	private var a_value:Double = 0.0
	private var b_value:Double = 0.0
	private var c_value:Double = 0.0
	@IBOutlet weak var editTextA:UITextField!
	@IBOutlet weak var editTextB:UITextField!
	@IBOutlet weak var editTextC:UITextField!
	@IBOutlet weak var x1TextView:UILabel!
	@IBOutlet weak var x2TextView:UILabel!

	override internal func viewDidLoad() {
		super.viewDidLoad()
		editTextA.removeTarget(nil, action: nil, for: .allEvents)
		editTextA.addTarget(self, action: #selector(targetMethod0(_:)), for: .editingChanged)
		editTextB.removeTarget(nil, action: nil, for: .allEvents)
		editTextB.addTarget(self, action: #selector(targetMethod1(_:)), for: .editingChanged)
		editTextC.removeTarget(nil, action: nil, for: .allEvents)
		editTextC.addTarget(self, action: #selector(targetMethod2(_:)), for: .editingChanged)
	}

	private func setNewValues() {
		var determinant:Double = pow(b_value,Double(2)) - 4 * a_value * c_value
		var x1:String = "", x2:String = ""
		if determinant > 0 {
					var temp1:Double = -b_value / (2 * a_value)
					var temp2:Double = sqrt(Double(determinant)) / (2 * a_value)
					x1 = String((temp1 + temp2))
					x2 = String(temp1 - temp2)
				} else if determinant == 0 {
					x1 = String(-b_value / (2 * a_value))
					x2 = x1
				} else {
					var temp1:Double = -b_value / (2 * a_value)
					var temp2:Double = sqrt(Double(-determinant)) / (2 * a_value)
					x1 = String(temp1) + "+i" + String(temp2)
					x2 = String(temp1) + "-i" + String(temp2)
				}
		x1TextView.text = x1
		x2TextView.text = x2
	}

	@objc func targetMethod0(_ editable:UITextField){
		if editable.text!.count != 0 {a_value = Double(editable.text!)!} else {
		a_value = 0
		}
		setNewValues()
	}
	@objc func targetMethod1(_ editable:UITextField){
		if editable.text!.count != 0 {b_value = Double(editable.text!)!} else {
		b_value = 0
		}
		setNewValues()
	}
	@objc func targetMethod2(_ editable:UITextField){
		if editable.text!.count != 0 {c_value = Double(editable.text!)!} else {
		c_value = 0
		}
		setNewValues()
	}
}